ledger/ledger-utxo/src/main/java/net/corda/v5/ledger/utxo/query/json/ContractStateVaultJsonFactory.java

```diff
@@ -0,0 +1,67 @@
+package net.corda.v5.ledger.utxo.query.json;
+
+import net.corda.v5.application.marshalling.JsonMarshallingService;
+import net.corda.v5.ledger.utxo.ContractState;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * The interface that needs to be implemented in order to create a JSON representation of a state type. Implementing
```

Comment:
Maybe go with something like:

```
Implement a {@link ContractStateVaultJsonFactory} to create a JSON representation of a state to store alongside the state in the vault. This JSON representation can be used to query states from flows using {@link UtxoLedgerService#query}.
```

Maybe add a `@see` for the query factory and the ledger service query method.